code/game/machinery/computer/camera_advanced.dm

```diff
@@ -0,0 +1,143 @@
+/obj/machinery/computer/camera_advanced
+	name = "advanced camera console"
+	desc = "Used to access the various cameras on the station."
+	icon_state = "cameras"
+	//circuit = /obj/item/weapon/circuitboard/security
+	var/mob/camera/aiEye/remote/eyeobj = new()
+	var/mob/living/carbon/human/current_user = null
+	var/list/networks = list("SS13")
+	var/datum/action/camera_off/off_action = new
+	var/datum/action/camera_jump/jump_action = new
+
+/obj/machinery/computer/camera_advanced/New()
+	eyeobj.origin = src
+
+/obj/machinery/computer/camera_advanced/check_eye(var/mob/user as mob)
+	if (get_dist(user, src) > 1 || user.eye_blind)
+		off_action.Activate()
+		return 0
+	return 1
+
+/obj/machinery/computer/camera_advanced/attack_hand(var/mob/user as mob)
+	if(..())
+		return
+	if(!iscarbon(user))
+		return
+	var/mob/living/carbon/L = user
+	if(!current_user)
+		off_action.target = user
+		off_action.Grant(user)
+		jump_action.target = user
+		jump_action.Grant(user)
+		current_user = user
+		eyeobj.user = user
+		eyeobj.name = "Camere Eye ([user.name])"
+		L.remote_view = 1
+		L.remote_control = eyeobj
+		L.client.perspective = EYE_PERSPECTIVE
+		if(!eyeobj.initialized)
+			for(var/obj/machinery/camera/C in cameranet.cameras)
+				if(!C.can_use())
+					continue
+				if(C.network&networks)
+					eyeobj.setLoc(get_turf(C))
+					break
+			eyeobj.initialized = 1
+	else
+		user << "The console is already in use!"
+
+/mob/camera/aiEye/remote
```

Comment:
Duplicated code with a lot of the code already being INHERITED by the aiEye. This is really poor code and should be reviewed.